### ISSUE TITLE
d3.v3.min.js location

### ISSUE THREAD

**Issue by jwelch95** [c3js/c3#445]
is d3.v3.min.js supposed to be located within the file, or is that meant to be downloaded separately. if so, where can I find a good download for it?

**Comment by PrashantMangale** [comment]
Hi I'm Prashant. I want to know that what is the difference between d3.js and d3.v3.min.js?
Are they different or d3.v3.min.js is a compressed version of d3.js.
If yes then can i use d3.js instead of d3.v3.min.js or vice-versa?
Please give me some feedback. Its a important for me to know exact difference between them,
Thanks in advance.

**Comment by aendrew** [comment]
@PrashantMangale The `.min.js` extension just says it's been "minified", i.e., all the spaces removed. Usually people also "uglify" it, which is an extension of minification wherein variables are renamed to be shorter and code is reorganised to be smaller. All this is intended to reduce the final file size of the library.

You can use them interchangeably though. Generally, I use the non-minified version while developing so that I can debug easier (you get things like full variable/function names and the ability to pinpoint which line an error is occurring on — in the minified version, every error is always just on "line 1" because all the line-breaks have been removed), and then I have a Gulp task that concatenates and minifies all of my JavaScript (libraries included) when I build a distributable ("dist") version.

tl;dr — if you have a build script that concatenates and minifies your JavaScript into a single file (I.e., you're using Yeoman), use the non-minified version. If you're just including C3/D3 from a CDN and/or don't have any need to debug the library itself, use the minified version.
